src/Magnum/LibOvrIntegration/Compositor.h

```diff
@@ -0,0 +1,217 @@
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::LibOvrIntegration::Compositor,
+ *        class @ref Magnum::LibOvrIntegration::Layer,
+ *        class @ref Magnum::LibOvrIntegration::EyeFovLayer,
+ *        enum @ref Magnum::LibOvrIntegration::LayerType
+ *
+ * @author Jonathan Hale (Squareys)
+ */
+
+#include <Magnum/Magnum.h>
+
+#include <bits/unique_ptr.h>
+#include <vector>
+
+#include "visibility.h"
+#include "LibOvrIntegration.h"
+
+#include <OVR/OVR_CAPI.h>
+
+namespace Magnum { namespace LibOvrIntegration {
+
+enum class LayerType: UnsignedByte {
+
+    /** @brief Described by ovrLayerEyeFov. */
+    EyeFov = ovrLayerType_EyeFov,
+
+    /** @brief Described by ovrLayerEyeFovDepth. */
+    EyeFovDepth = ovrLayerType_EyeFovDepth,
+
+    /** @brief Described by ovrLayerQuad. */
+    QuadInWorld = ovrLayerType_QuadInWorld,
+
+    /** @brief Described by ovrLayerQuad. Displayed in front of your face, moving with the head. */
+    QuadHeadLocked = ovrLayerType_QuadHeadLocked,
+
+    /** @brief Described by ovrLayerDirect. Passthrough for debugging and custom rendering. */
+    Direct = ovrLayerType_Direct
+
+};
+
+/**
+ * @brief Wrapper around an ovrLayerHeader.
+ *
+ * If you need to be able to change the layer, use one of the type specific
+ * layer classes instead: @ref LayerEyeFov
+ *
+ * @author Jonathan Hale (Squareys)
+ */
+class MAGNUM_LibOvrIntegration_EXPORT Layer {
+public:
+    /**
+     * @brief Contructor.
+     * @param type Type of this layer.
+     */
+    Layer(const LayerType type);
+
+    /**
+     * @brief Set whether to process this layer in high quality.
+     *
+     * High quality mode costs performance, but looks better.
+     *
+     * @param b If true, layer will be in high quality mode, if false, the
+     *          layer will be in lower quality mode.
+     * @return Reference to self (for method chaining)
+     */
+    Layer& setHighQuality(bool highQuality) {
+        if(highQuality) {
+            _layer.Header.Flags |= ovrLayerFlag_HighQuality;
+        } else {
+            _layer.Header.Flags &= ~ovrLayerFlag_HighQuality;
+        }
+        return *this;
+    }
+
+    /**
+     * @brief Enable/disable a the layer.
+     * @param enabled If true, the layer will be enabled.
+     * @return Reference to self (for method chaining)
+     */
+    Layer& setEnabled(bool enabled);
+
+    /** @brief Type of this layer. */
+    LayerType layerType() const {
+        return _TYPE;
+    }
+
+    /** @brief The underlying ovrLayerHeader. */
+    const ovrLayerHeader& layerHeader() const {
+        return _layer.Header;
+    }
+
+protected:
+    ovrLayer_Union _layer;
```

Comment:
(This is also rather QtCreator's fault -- I was unable to persuade it to do otherwise myself.)

I'm indenting the `public:`/`private:` access specifiers with 4 spaces and the contents inside with another 4 spaces (it might be wasteful, but I did that for whole project, so just for consistency).